lib/hobo_fields/model.rb

```diff
@@ -117,6 +117,7 @@ def self.belongs_to_with_field_declarations(name, *args, &block)
       end
       column_options = {}
       column_options[:null] = options.delete(:null) || false
+      column_options[:limit] = options.delete(:limit) if options.has_key?(:limit)
```

Comment:
> So the programmer will have to remember that any FKs to a bigint are themselves bigints. There's no chance of us inferring that?

We will infer it, I put this in as a helper to migrate existing FKs. `belongs_to` didn't support a limit before, so now you can specify it to migrate old to new.

Otherwise, any *new* FKs will default to a bigint

Reply:
But correct - if we change a tables PK to be a bigint, the onus is still on the developer to catch the places where the FKs pointing to it are not bigints. 

A walker seems like a good idea...if we could build a mass schema diagram programmatically such a thing shouldn't be _too_ hard to do

Reply:
Biking in this morning I thought of something scary... it's possible that MySQL won't do proper joins if there's a mixture of INT and BIGINT as the join key. Maybe we can devise a test to convince ourselves they do it fine?
Here's an example of the potential problem:
```
explain select * from calls where id = 781444924;
+----+-------------+-------+------------+-------+---------------+---------+---------+-------+------+----------+-------+
| id | select_type | table | partitions | type  | possible_keys | key     | key_len | ref   | rows | filtered | Extra |
+----+-------------+-------+------------+-------+---------------+---------+---------+-------+------+----------+-------+
|  1 | SIMPLE      | calls | NULL       | const | PRIMARY       | PRIMARY | 4       | const |    1 |   100.00 | NULL  |
+----+-------------+-------+------------+-------+---------------+---------+---------+-------+------+----------+-------+
```
So that was a TABLE SEEK like we'd expect. But if we make them change the type of the key:
```
explain select * from calls where cast(id as unsigned) = 781444924;
+----+-------------+-------+------------+------+---------------+------+---------+------+-----------+----------+-------------+
| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows      | filtered | Extra       |
+----+-------------+-------+------------+------+---------------+------+---------+------+-----------+----------+-------------+
|  1 | SIMPLE      | calls | NULL       | ALL  | NULL          | NULL | NULL    | NULL | 271787157 |   100.00 | Using where |
+----+-------------+-------+------------+------+---------------+------+---------+------+-----------+----------+-------------+
```
^ They give up and do a TABLE SCAN!